### ISSUE TITLE
Information about transforms

### ISSUE THREAD

**Issue by VGrondin** [vc-bonn/torchhull#24]
Hi, first off, thanks for sharing the repository it seems powerfull! 

I have been trying to get it to work for a while, but every time, the output hull is empty. I suspect I am not using the correct transform so if you could double check with me that would be great. 

Here is what I am using:
```
# Intrinsic matrix (3x3)
K = np.array([
    [fl_x, 0, cx],
    [0, fl_y, cy],
    [0,    0, 1]
], dtype=np.float32)

perspective_M = K_to_perspective(K, dtype=torch.float32, device=DEVICE)
print(f"Perspective matrix:\n{perspective_M}")

# Extrinsic world-to-camera (4x4), OpenGL format
T = extrinsic_world_to_cam.astype(np.float32)
# To torch
T = torch.from_numpy(T).to(dtype=torch.float32, device=DEVICE)
print(f"Extrinsic (world-to-camera) matrix:\n{T}")

# 4x4 fused projection matrix
KT = perspective_M @ T
print(f"Projection matrix (K @ T):\n{KT}")
```

Console output:
```
Perspective matrix:
tensor([[ 1.4000,  0.0000,  0.0000,  0.0000],
        [ 0.0000,  1.8667,  0.0000,  0.0000],
        [ 0.0000,  0.0000, -1.0002, -0.2000],
        [ 0.0000,  0.0000, -1.0000,  0.0000]], device='cuda:0')
Extrinsic (world-to-camera) matrix:
tensor([[  0.8574,  -0.5111,  -0.0611,   1.1787],
        [ -0.4985,  -0.7949,  -0.3458, -13.3134],
        [  0.1282,   0.3269,  -0.9363,   9.6335],
        [  0.0000,   0.0000,   0.0000,   1.0000]], device='cuda:0')
Projection matrix (K @ T):
[[  1.200303    -0.7155185   -0.08547416   1.6501449 ]
 [ -0.93052995  -1.4838829   -0.64548457 -24.851648  ]
 [ -0.12822303  -0.32697105   0.9365089   -9.835403  ]
 [ -0.12819739  -0.32690567   0.9363216   -9.633456  ]]
```

The perspective matrix is computed like this:
```
fovy = 2 * np.atan(7.2 / (2 * 6.72))    # 52 deg
aspect = 4032 / 3024
torch.tensor(
        [
            [1.0 / (np.tan(fovy / 2.0) * aspect), 0.0, 0.0, 0.0],
            [0.0, 1.0 / np.tan(fovy / 2.0), 0.0, 0.0],
            [0.0, 0.0, -(far + near) / (far - near), -(2.0 * far * near) / (far - near)],
            [0.0, 0.0, -1.0, 0.0],
        ],
        dtype=dtype,
        device=device,
    )
```

After stacking > 3 views, the transform input to torchhull is shaped [3, 4, 4]
The input masks are also well aligned (same object). 
Any idea why it is not working?

Best,
Vincent

**Comment by stotko** [comment]
It looks like the conversion from OpenCV to OpenGL convention is not fully correct, i.e. incomplete. The 4x4 projection matrix looks fine, but the extrinsics are still provided in OpenCV convention, right? This means that your extrinsics expects (and outputs) an OpenCV point wheras the projection expects an OpenGL point. To fix this, you would need to convert between conventions, i.e. flip the y and z axes. This can be done by (matrix-)multiplying `torch.tensor([1, -1, -1, 1]).diag()` from both sides to the extrinsics.

Alternatively, you can give the current dev version from the main branch a try. It has an option to specify the convention, so you could directly pass in the fused matrix in OpenCV convention.

**Comment by VGrondin** [comment]
Hi @stotko, thanks for replying!

I believe the extrinsic are in OpenGL convention since I use them to position the camera in Open3d, which convention is the same as OpenGL (but the z axis might be looking at the camera instead of aways from the camera, if that changes anything).

**Comment by VGrondin** [comment]
Hi Patrick, as you pointed out in your last reply, the issue was indeed to also convert the segmentation mask into OpenGL convention. It works now! Many thanks!
